src/ng/directive/ngRepeat.js

```diff
@@ -652,6 +650,12 @@ var ngRepeatDirective = ['$parse', '$animate', '$compile', function($parse, $ani
             }
           }
           lastBlockMap = nextBlockMap;
+
+          // Clear the value property from the hashFnLocals object to prevent a reference to the last value
+          // being leaked into this ngRepeatCompile function scope
+          if (hashFnLocals) {
+            hashFnLocals[valueIdentifier] = undefined;
```

Comment:
What about `keyIdentifier`? Can't is be non-primitive?

Reply:
The `keyIdentifier` value is either an index or the `itemKey` from `for (var itemKey in collection)`. So it should always be a number or string.